aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -118,18 +118,15 @@ Completable hydrate() {
                         LOG.warn("Initial cloud sync failed.", failureToSync);
                         DataStoreErrorHandler dataStoreErrorHandler =
                             dataStoreConfigurationProvider.getConfiguration().getDataStoreErrorHandler();
-                        if (dataStoreErrorHandler != null) {
-                            dataStoreErrorHandler.accept(
-                                new DataStoreException(
-                                    "Initial cloud sync failed.",
-                                    failureToSync,
-                                    "Check your internet connection."));
-                        }
+                        dataStoreErrorHandler.accept(
+                            new DataStoreException(
+                                "Initial cloud sync failed.",
+                                failureToSync,
+                                "Check your internet connection."));
                     })
                     .doOnComplete(() ->
                         LOG.info("Successfully sync'd down model state from cloud.")
                     )
-                    .onErrorComplete()
```

Comment:
I need the `hydrate()` call to surface its failures, so that the orchestrator can respond to them by cleaning up and shutting down.